### PR TITLE
add uniform processors for altclip + chinese_clip

### PR DESCRIPTION
**Pull request by molbap**
# What does this PR do?

Adds two models for https://github.com/huggingface/transformers/pull/30511 , see parent PR for more details

### CODE REVIEW

**Comment by molbap** [comment]
Tests will succeed when #31197 is merged. See #30511 tests to confirm.

**Review by amyeroberts — COMMENTED** [review]
Thanks for working on this! 

Overall looks good to me. Main thought is that we probably want to move a lot of the kwarg preparation logic outside of the main `__call__` method, and it can probably be abstracted into something more general for most processors. 

Regarding tests, it would be good to check defaults and updates are correctly passed for all of the possible kwargs, particularly defaulting to the tokenizer.init_kwargs and ensuring these can be overridden. 

Would be good to get a second opinion from @qubvel here!

**Comment by molbap** [comment]
Thanks! I'm currently moving the kwargs merging in a common method actually, and explaining order of operations wrt to kwargs priorities.

**Review by qubvel — COMMENTED** [review]
Great work 🤗  Very structured and clean code!
I have the same concern as @amyeroberts that the logic of merging is worth split/move to separate methods. It is a bit hard to follow merging steps of different kwargs.

What do you think about grouping merging strategy per modality instead of per step?
not sure if its possible, because some kwargs, probably, have to be popped in advance.

Something like this:
```python
images_kwargs = {} if images_kwargs is None else images_kwargs
default_images_kwargs = ChineseClipProcessorKwargs._defaults.get("images_kwargs", {}).copy()

# merging vision kwargs
images_kwargs = {**default_images_kwargs, **images_kwargs}   # with default
images_kwargs = {**images_kwargs, **kwargs.pop("images_kwargs", {})}  # with passed dict
images_kwargs = merge_with_kwargs_by_key(images_kwargs, ...)
images_kwargs = merge_with_common(images_kwargs, common_kwargs)
```

File: src/transformers/processing_utils.py
Comment (molbap) [reply src/transformers/processing_utils.py:172]: sure! adding it

Comment (molbap) [reply src/transformers/processing_utils.py:172]: for your comment above, kwargs merging is broken down in https://github.com/huggingface/transformers/pull/31197 in a common method, that's a bit more up-to-date (apologies for the split PRs)

File: tests/models/altclip/test_processor_altclip.py
Comment (molbap) [reply tests/models/altclip/test_processor_altclip.py:1]: Moving it to a common testing instance!

**Comment by molbap** [comment]
I simplified the addition/testing of uniformized processors on that one. It's pulling changes from #31197 so the file diff will be reduced once that first one is merged! shoud be ready for review after that.

**Comment by molbap** [comment]
This one should be ready for review @amyeroberts after #31197 is reviewed&merged!

File: src/transformers/models/chinese_clip/processing_chinese_clip.py
Comment (amyeroberts) [inline src/transformers/models/chinese_clip/processing_chinese_clip.py:26]: nit 
```suggestion
from ...processing_utils import ProcessingKwargs, ProcessorMixin
```

File: tests/models/altclip/test_processor_altclip.py
Comment (amyeroberts) [inline tests/models/altclip/test_processor_altclip.py:1]: Missing copyright header

File: tests/models/altclip/test_processor_altclip.py
Comment (amyeroberts) [inline tests/models/altclip/test_processor_altclip.py:37]: What's the reason for the pass here? I'm guessing the parent tearDown is removing objects which aren't present for this tester?

File: tests/test_processing_common.py
Comment (amyeroberts) [inline tests/test_processing_common.py:132]: Does this mean the padding behaviour of the processors isn't backwards compatible i.e. it used to default to `"max_length"` and doesn't anymore? Or is this because of the new processors being tested?

File: tests/test_processing_common.py
Comment (amyeroberts) [inline tests/test_processing_common.py:66]: Why do we need the try/except patten here?

**Review by amyeroberts — COMMENTED** [review]
Thanks for working on this! 

Just some small comments / questions

File: tests/test_processing_common.py
Comment (molbap) [reply tests/test_processing_common.py:66]: because some processors are initialized with a self.tmpdirname, some don't have this attribute but have a self.model_id  instead - I can check for attribute presence/absence instead of try/except, it'd be cleaner

File: tests/test_processing_common.py
Comment (molbap) [reply tests/test_processing_common.py:132]: the latter, this is just because of new processors being tested - no weird BC behaviour here as far as I know

File: tests/models/altclip/test_processor_altclip.py
Comment (molbap) [reply tests/models/altclip/test_processor_altclip.py:37]: initially, yes, but now it's just not used :grimacing: will remove

File: src/transformers/models/altclip/processing_altclip.py
Comment (molbap) [reply src/transformers/models/altclip/processing_altclip.py:69]: ah true, will try that, should do nicely!

**Comment by molbap** [comment]
@amyeroberts I forgot to launch slow tests, they are a bit slow, but should be good for review now!
- removed try/except block for processors attributes
- formatting
- removed unused teardown method

File: src/transformers/models/altclip/processing_altclip.py
Comment (amyeroberts) [inline src/transformers/models/altclip/processing_altclip.py:62]: Given we don't advertise `audio` is an input when calling the processor, do we want to add `feature_extractor` here to the docstring?

File: src/transformers/models/altclip/processing_altclip.py
Comment (amyeroberts) [inline src/transformers/models/altclip/processing_altclip.py:37]: Do we have an issue open for this, so that we don't forget?

File: src/transformers/models/altclip/processing_altclip.py
Comment (amyeroberts) [inline src/transformers/models/altclip/processing_altclip.py:126]: Do you think we should have an equivalent `init_kwargs` for the image processors and feature extractors for consistency? Or better to leave just in one place?

File: src/transformers/models/altclip/processing_altclip.py
Comment (amyeroberts) [inline src/transformers/models/altclip/processing_altclip.py:25]: Rather than try/except could we do this in a more explicity way e.g. by testing on the version in the env?

File: tests/models/altclip/test_processor_altclip.py
Comment (amyeroberts) [reply tests/models/altclip/test_processor_altclip.py:1]: I think this needs to be moved to under the altclip model folder

**Review by amyeroberts — COMMENTED** [review]
Looks great - just a few small things here and there! 

cc @ArthurZucker for reference on the changes in processors and their kwargs

File: src/transformers/models/altclip/processing_altclip.py
Comment (molbap) [reply src/transformers/models/altclip/processing_altclip.py:62]: Was hesitant on that - wdyt about not advertising unused modalities, rather? I wanted to have the same signature everywhere but that might be overkill

File: src/transformers/models/altclip/processing_altclip.py
Comment (molbap) [reply src/transformers/models/altclip/processing_altclip.py:37]: ah, this is outdated - with the new import structure of Unpack this issue is not there anymore!

File: src/transformers/models/altclip/processing_altclip.py
Comment (molbap) [reply src/transformers/models/altclip/processing_altclip.py:126]: currently the image processors and other extractors don't need this to work (this is mostly for fast tokenizers), not opposed to including it as well but it seems to be more a tokenizers thing?

File: tests/models/altclip/test_processor_altclip.py
Comment (molbap) [reply tests/models/altclip/test_processor_altclip.py:1]: I'm not sure I understand why? Is it because there was no test before?

File: src/transformers/models/altclip/processing_altclip.py
Comment (molbap) [reply src/transformers/models/altclip/processing_altclip.py:70]: done!

File: src/transformers/models/altclip/processing_altclip.py
Comment (amyeroberts) [reply src/transformers/models/altclip/processing_altclip.py:126]: OK, let's leave for now!

File: src/transformers/models/altclip/processing_altclip.py
Comment (molbap) [reply src/transformers/models/altclip/processing_altclip.py]: done!

File: tests/models/altclip/test_processor_altclip.py
Comment (amyeroberts) [reply tests/models/altclip/test_processor_altclip.py:1]: Ah, I think I misunderstood your previous comment: I thought you meant move this file under a higher level e.g. `tests/test_processor_xxx.py` or something

File: src/transformers/models/altclip/processing_altclip.py
Comment (molbap) [reply src/transformers/models/altclip/processing_altclip.py:55]: oh, no I understand the issue here @amyeroberts , in this case `feature_extractor` is unfortunately a backwards compatible name for the image processor. So we cannot remove it.

Comment (amyeroberts) [reply src/transformers/models/altclip/processing_altclip.py:55]: Yes, I think we need to be able to accept feature_extractors as an input, but I don't think they should be added to the docstring as it's confusing. This probably falls under the same scope for @qubvel's deprecation decorators again

File: src/transformers/models/altclip/processing_altclip.py
Comment (qubvel) [inline src/transformers/models/altclip/processing_altclip.py:63]: As soon as we added `@deprecate_kwarg(old_name="feature_extractor", version="5.0.0", new_name="image_processor")` we no longer need `feature_extractor` in the function signature, an input keyword argument `feature_extractor` will be renamed to `image_processor`.

```suggestion
    def __init__(self, image_processor=None, tokenizer=None):
```

Comment (molbap) [reply src/transformers/models/altclip/processing_altclip.py:63]: yes indeed, thanks. should be good to go!

File: src/transformers/models/altclip/processing_altclip.py
Comment (molbap) [reply src/transformers/models/altclip/processing_altclip.py]: agreed! done!

**Comment by molbap** [comment]
@amyeroberts I think this one is ok for a review!

File: src/transformers/models/altclip/processing_altclip.py
Comment (amyeroberts) [inline src/transformers/models/altclip/processing_altclip.py:26]: Similar to what we have with `PILImageResampling` could we define an object, where we do this importing once in one place and then import from there?

**Review by amyeroberts — APPROVED** [review]
Nice! Thanks for all the work on this 🤗

File: src/transformers/models/altclip/processing_altclip.py
Comment (molbap) [reply src/transformers/models/altclip/processing_altclip.py:26]: ah that's a good point! will do that

**Comment by molbap** [comment]
Re-pinging @amyeroberts on that one and @qubvel and @zucchini-nlp optionally if you want to take another look - not much changed, just was between sprints and didn't merge that one :grimacing: should be good to go now (I know you approved, Amy, but that was a while ago!)

**Comment by molbap** [comment]
Tests that were working are broken since passing from `crop_size` to `size` in common tests I believe  - will look into this soon

File: tests/models/altclip/test_processor_altclip.py
Comment (amyeroberts) [inline tests/models/altclip/test_processor_altclip.py:65]: No need to add this now -- it's in athe processormixin

**Review by amyeroberts — APPROVED** [review]
Thanks for adding!

File: tests/models/altclip/test_processor_altclip.py
Comment (molbap) [reply tests/models/altclip/test_processor_altclip.py:65]: right, done!

**Pull request merged**
